Plugins/Tweaks/Tweaks/PreserveDepletedItems.cpp

```diff
@@ -0,0 +1,44 @@
+#include "Tweaks/PreserveDepletedItems.hpp"
+#include "API/CNWSCreature.hpp"
+#include "API/CNWSItem.hpp"
+#include "API/CNWSObjectActionNode.hpp"
+#include "API/Functions.hpp"
+#include "API/Globals.hpp"
+#include "API/Version.hpp"
+
+#include "Services/Hooks/Hooks.hpp"
+#include "Utils.hpp"
+
+
+namespace Tweaks {
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+
+NWNXLib::Hooking::FunctionHook* PreserveDepletedItems::pAIActionItemCastSpell_hook;
+PreserveDepletedItems::PreserveDepletedItems(ViewPtr<Services::HooksProxy> hooker)
+{
+    hooker->RequestExclusiveHook<Functions::CNWSCreature__AIActionItemCastSpell>
+                                    (&CNWSCreature__AIActionItemCastSpell_hook);
+
+    pAIActionItemCastSpell_hook = hooker->FindHookByAddress(Functions::CNWSCreature__AIActionItemCastSpell);
+}
+
+
+uint32_t PreserveDepletedItems::CNWSCreature__AIActionItemCastSpell_hook(CNWSCreature *pThis, CNWSObjectActionNode *pNode)
+{
+    // If at risk of destroying the item, inflate charge count temporarily to bypass the destroy
+    // event, then set it back to mark the spells as unusable.
+    auto *pItem = Utils::AsNWSItem(Utils::GetGameObject((Types::ObjectID)pNode->m_pParameter[0]));
+    if (pItem && pItem->m_nNumCharges <= 5)
```

Comment:
Why 5? What if the item consumes more than 5 charges on use? (for example, an item with multiple spell levels on it?) Is 5 the most charges you can consume at once?

Reply:
Yes, there's no in-game property to make it cost more than 5 charges per use. This is hardcoded, so simple CC can't bypass.